cmake/thirdParties.cmake

```diff
@@ -0,0 +1,25 @@
+if(MATIO_WITH_HDF5)
+    find_package(HDF5)
```

Comment:
On my Win machine with CMake 3.13.3 installed, when I configure CMake to generate for VS Win64, it finds the HDF5 libs for 32-bit (likely because they appear first on `%PATH%: C:\Program Files (x86)\HDF_Group\HDF5\1.8.13\bin;C:\Program Files\HDF_Group\HDF5\1.8.13\bin`).

* Is there any chance to check if platform matches?
* Is there any chance to check if HDF5 >= 1.8.x is satisfied?

Reply:
> On my Win machine with CMake 3.13.3 installed, when I configure CMake to generate for VS Win64, it finds the HDF5 libs for 32-bit (likely because they appear first on `%PATH%: C:\Program Files (x86)\HDF_Group\HDF5\1.8.13\bin;C:\Program Files\HDF_Group\HDF5\1.8.13\bin`).
> 
> * Is there any chance to check if platform matches? There are some discussions going on on SO or the HDF5 forum about that issue, but so far I have not figured out a proper solution or workaround.

I see that the HDF5 1.10 and 1.12 cmake files set `PACKAGE_VERSION_UNSUITABLE` when the package has the wrong bit-ness. But I don't know if it continuous searching for a correct version. 1.8 does not have this. If you provide a CMake argument `-DHDF_DIR="dir"`, it should check this directory first, see https://cmake.org/cmake/help/latest/command/find_package.html for search order.

> * Is there any chance to check if HDF5 >= 1.8.x is satisfied?

Yes, seems you found it as well.

> * There is no option if the shared or static HDF5/Zlib library is to be found. But that is probably too picky as it requires new CMake options for matio, too. It even might get messy on Win with /MT oder /MD settings required to be consistent,

With HDF5 1.10 and 1.12 you might be able to specify it as component, e.g. `find_package(HDF5 1.10 COMPONENTS shared)`, but I only checked the `hdf5-config.cmake`, I have not tried it. 
And indeed matio would need an option to specify this. I don't think configure has this?